app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -295,6 +333,14 @@ def ca_cert
 
     private
 
+    def os_name_mapping(host)
+      host.operatingsystem.name =~ /redhat|centos/i ? 'rhel': host.operatingsystem.name
+    end
+
+    def arch_name_mapping(host)
+      host.architecture.name == 'x86_64' ? 'x64' : host.architecture.name if host.architecture
```

Comment:
My point is - if the OS is in `@host` , the architecture must be there as well, as it's a prerequisite, so why checking for its existence